pkg/runtime/grpc/instances.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpcruntime
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/runtime"
+)
+
+func (r *Runtime) RemoveGadgetInstance(ctx context.Context, runtimeParams *params.Params, id string) error {
+	conn, err := r.getConnToRandomTarget(ctx, runtimeParams)
+	if err != nil {
+		return err
+	}
```

Comment:
How does it work when using ig daemon and multiple servers? I suppose in that case we need to contact all of them.

Reply:
Yeah, exactly. I thought I had implemented it that way already in this PR, maybe it got dropped with the numerous refactorings. I will re-check.

Reply:
Fixed.